### ISSUE TITLE
__resolveType should be optional in interface resolvers

### ISSUE THREAD

**Issue by Akryum** [dotansimha/graphql-code-generator#3094]
**Describe the bug**
 
Currently, `Resolvers` interface for GraphQL Interface types requires the `__resolveType` resolver to be defined. This is an issue in any project that split resolvers in multiple files: `__resolveType` doesn't need to be implemented each time a resolvers object is declared for the GraphQL Interface.

**To Reproduce**

1. My GraphQL schema:

```graphql
interface Animal {
  id: ID!
  computedField: String
}

type Cat implements Animal {
  id: ID!
  computedField: String
}

type Dog implements Animal {
  id: ID!
  computedField: String
}
```

2. Resolvers:

File A:

```ts
export const resolvers: Resolvers = {
  Animal: {
    __resolveType: (item) => item.type,
  },
}
```

File B:

```ts
export const resolvers: Resolvers = {
  // Error thrown here: Property '__resolveType' is missing
  Animal: {
    computedField: (item) => item.foo + 'bar',
  },
}
```

Indeed, in the generated types, `__resolveType` is always required in the resolvers object:

```ts
export type AnimalResolvers = {
  __resolveType: TypeResolveFn
}
```

**Expected behavior**

No error in File B.

The generated type should look like:

```ts
export type AnimalResolvers = {
  __resolveType?: TypeResolveFn
}
```

**Environment:**

- OS: Linux
-    "@graphql-codegen/cli": "^1.9.1",
-    "@graphql-codegen/typescript": "^1.9.1",
-    "@graphql-codegen/typescript-operations": "^1.9.1",
-    "@graphql-codegen/typescript-resolvers": "^1.9.1",
-    "@graphql-codegen/fragment-matcher": "^1.9.1",
-    "@graphql-codegen/introspection": "^1.9.1",
- NodeJS: 12.13.0

**Additional context**

**Comment by dotansimha** [comment]
@Akryum I added this in this PR: https://github.com/dotansimha/graphql-code-generator/pull/3123

Can you please try the alpha version? `1.9.2-alpha-976a8ac6.41`
You should be able to set now `optionalResolveType: true` in your config, and it will generate `__resolveType` as optional field for both `union` and `interface` resolvers.

**Comment by dotansimha** [comment]
Fixed in `v1.10.0`.
